File: api/equity-regulatory-reporting/equity-regulatory-reporting.Domain/Entities/DocumentType.cs
using equity_regulatory_reporting.Domain.Common;

namespace equity_regulatory_reporting.Domain.Entities;

public class DocumentType : AuditableEntity
{
    public string Name { get; set; } = string.Empty;
    public string Abbreviation { get; set; } = string.Empty;
    public string? ValidationRegex { get; set; }

    public ICollection<DocumentTypePersonType> AllowedPersonTypes { get; set; } = [];
    public ICollection<Person> Persons { get; set; } = [];
}

File: app/equity-regulatory-reporting-app/src/types/document-type.ts
import type { PersonTypeValue } from "@/lib/person-types";

export interface DocumentTypeDto {
  id: string;
  name: string;
  abbreviation: string;
  validationRegex: string | null;
  allowedPersonTypes: PersonTypeValue[];
}

export interface CreateDocumentTypeDto {
  name: string;
  abbreviation: string;
  validationRegex: string | null;
  allowedPersonTypes: PersonTypeValue[];
}

export type UpdateDocumentTypeDto = CreateDocumentTypeDto;
